runtime/interpreter/value.go

```diff
@@ -12651,7 +12705,22 @@ func (v *DictionaryValue) Transfer(
 	}
 
 	if isResourceKinded {
+		// Update the resource in-place,
+		// and also update all values that are referencing the same value
+		// (but currently point to an outdated Go instance of the value)
+
 		v.dictionary = dictionary
+
+		newStorageID := dictionary.StorageID()
+
+		interpreter.updateReferencedResource(
+			currentStorageID,
+			newStorageID,
+			func(value ReferenceTrackedResourceKindedValue) {
+				value.(*DictionaryValue).dictionary = dictionary
```

Comment:
same question as above

Reply:
Fixed in 7295252fe948a5fa81f01fdd8c574234493a67a4